akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphUnzipWithSpec.scala

```diff
@@ -0,0 +1,281 @@
+      rightSubscription.request(2)
+
+      leftProbe.expectNext(6)
+      leftProbe.expectNoMsg(100.millis)
+
+      rightProbe.expectNext("2+2")
+      rightProbe.expectNext("3+3")
+      rightProbe.expectNoMsg(100.millis)
+
+      leftSubscription.request(1)
+      rightSubscription.request(1)
+
+      leftProbe.expectNext(8)
+      rightProbe.expectNext("4+4")
+
+      leftProbe.expectComplete()
+      rightProbe.expectComplete()
+    }
+
+    "work in the sad case" in {
+      val leftProbe = TestSubscriber.manualProbe[LeftOutput]()
+      val rightProbe = TestSubscriber.manualProbe[RightOutput]()
+
+      FlowGraph.closed() { implicit b ⇒
+        val unzip = b.add(UnzipWith[Int, Int, String](b ⇒ (1 / b, 1 + "/" + b)))
+
+        Source(-2 to 2) ~> unzip.in
+
+        unzip.out0 ~> Flow[LeftOutput].buffer(4, OverflowStrategy.backpressure) ~> Sink(leftProbe)
+        unzip.out1 ~> Flow[RightOutput].buffer(4, OverflowStrategy.backpressure) ~> Sink(rightProbe)
+      }.run()
+
+      val leftSubscription = leftProbe.expectSubscription()
+      val rightSubscription = rightProbe.expectSubscription()
+
+      leftSubscription.request(2)
+      leftProbe.expectNext(1 / -2)
+      leftProbe.expectNext(1 / -1)
+
+      rightSubscription.request(1)
+      rightProbe.expectNext("1/-2")
+
+      leftSubscription.request(2)
+      leftProbe.expectError() match {
+        case a: java.lang.ArithmeticException ⇒ a.getMessage should be("/ by zero")
+      }
+      rightProbe.expectError()
+
+      leftProbe.expectNoMsg(100.millis)
+      rightProbe.expectNoMsg(100.millis)
+    }
+
+    "unzipWith expanded Person.unapply (3 ouputs)" in {
+      val probe0 = TestSubscriber.manualProbe[String]()
+      val probe1 = TestSubscriber.manualProbe[String]()
+      val probe2 = TestSubscriber.manualProbe[Int]()
+
+      case class Person(name: String, surname: String, int: Int)
+
+      FlowGraph.closed() { implicit b ⇒
+        val unzip = b.add(UnzipWith((a: Person) ⇒ Person.unapply(a).get))
+
+        Source.single(Person("Caplin", "Capybara", 3)) ~> unzip.in
+
+        unzip.out0 ~> Sink(probe0)
+        unzip.out1 ~> Sink(probe1)
+        unzip.out2 ~> Sink(probe2)
+      }.run()
+
+      val subscription0 = probe0.expectSubscription()
+      val subscription1 = probe1.expectSubscription()
+      val subscription2 = probe2.expectSubscription()
+
+      subscription0.request(1)
+      subscription1.request(1)
+      subscription2.request(1)
+
+      probe0.expectNext("Caplin")
+      probe1.expectNext("Capybara")
+      probe2.expectNext(3)
+
+      probe0.expectComplete()
+      probe1.expectComplete()
+      probe2.expectComplete()
+    }
+
+    "work with up to 20 outputs" in {
+      val probe0 = TestSubscriber.manualProbe[Int]()
+      val probe5 = TestSubscriber.manualProbe[String]()
+      val probe10 = TestSubscriber.manualProbe[Int]()
+      val probe15 = TestSubscriber.manualProbe[String]()
+      val probe19 = TestSubscriber.manualProbe[String]()
```

Comment:
This is testing a Tuple10, so we have not covered the "farthest" case, can you change this test to cover up to 20 inputs? (In theory Unzip could have more than this, it's zipWith which is limiting us to it, so let's be symmetric and also do 20 here).